binary_search_tree/binary_search_tree.py

```diff
@@ -1,14 +1,49 @@
 class BinarySearchTree:
-  def __init__(self, value):
-    self.value = value
-    self.left = None
-    self.right = None
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
 
-  def insert(self, value):
-    pass
+    def insert(self, value):
 
-  def contains(self, target):
-    pass
+        current = self
 
-  def get_max(self):
-    pass
+        # loop through tree until None is reached
+        while True:
+            # if value is less than the Node value, go left
+            if value < self.value:
+
+                if not self.left:
+                    self.left = BinarySearchTree(value)
+                    break
+                self = self.left
+
+            # if value is same or bigger than the Node value, go right
+            else:
+
+                if not self.right:
+                    self.right = BinarySearchTree(value)
+                    break
+                self = self.right
+
+    def contains(self, target):
+        current = self
+        # while loop until there is no more Node
+        while self:
```

Comment:
I am not sure if this loop will stop iterating since self is always truthy.  To traverse through the left and right branch of the head node, you may want to consider using recursion

Reply:
in the while loop, I set self = self.right (or self.left) so self becomes None which ends loop